include/mem/mem.h

```diff
@@ -113,6 +114,14 @@ namespace mem
         typename std::enable_if<!std::is_reference<T>::value, typename std::add_lvalue_reference<T>::type>::type
         rcast() & noexcept;
 
+        template <typename Container>
+        constexpr pointer put_bytes(const Container& obj) const noexcept {
+            using value_t = typename Container::value_type;
+            constexpr auto size_e  = sizeof(value_t);
+            region(*this, size_e * obj.size()).copy(obj.data());
```

Comment:
Your use-case suggests using this to patch a program, but most programs don't have RWX code segments.

Also `mem::pattern("b8 01 00 00 00 c3").bytes()` returns a pointer so your example wouldn't compile.

Reply:
What I did, was add a `.data()` method to `mem::pattern` that returns the same as `bytes()` and since it already had a `size` method, I can just pass a pattern directly.